### PR TITLE
Upgrade iris pregen & use toolbelt

### PR DESCRIPTION
**Pull request by svdgoor**

### CODE REVIEW

**Review by cyberpwnn — COMMENTED** [review]
(ignore this review, github buggy)

File: src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregenCreate.java [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+import org.bukkit.World;
+
+import java.util.Arrays;
+
+
+public class CommandIrisPregenCreate extends MortarCommand {
+
+    private static final KList<String> argus = new KList<>("radius=", "width=", "height=", "x=", "z=");
+
+    public CommandIrisPregenCreate() {
```
Comment (cyberpwnn) [inline src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregenCreate.java]: This should probably be "start" instead of "create". Same for the command name

File: src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregenCreate.java [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+                if (IrisToolbelt.isIrisWorld(w)) {
+                    list.add("world=" + w.getName());
+                }
+            });
+        }
+    }
+
+    @Override
+    protected String getArgsUsage() {
+        return "<radius> [width=<width>] [height=<height>] [x=<centerX>] [z=<centerZ>] [world=<world>] [-here]";
```
Comment (cyberpwnn) [inline src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregenCreate.java]: The width & height setting bothers me a lot. Im not confident it's fully supported. It would be safe to rely on a size only. That and no one will really use this as generally pregens are symetrical. And ... with this they now have to type out two long arguments instead of just a size.

File: src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregenToggle.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+import com.volmit.iris.Iris;
+import com.volmit.iris.core.gui.PregeneratorJob;
+import com.volmit.iris.util.collection.KList;
+import com.volmit.iris.util.plugin.MortarCommand;
+import com.volmit.iris.util.plugin.VolmitSender;
+
+public class CommandIrisPregenToggle extends MortarCommand {
+
+    public CommandIrisPregenToggle() {
```
Comment (cyberpwnn) [inline src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregenToggle.java]: Pause should be the default alias (and class name)

File: src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregen.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+package com.volmit.iris.core.command.pregen;
+
+import com.volmit.iris.Iris;
+import com.volmit.iris.core.tools.IrisToolbelt;
+import com.volmit.iris.util.collection.KList;
+import com.volmit.iris.util.plugin.Command;
+import com.volmit.iris.util.plugin.MortarCommand;
+import com.volmit.iris.util.plugin.VolmitSender;
+
+public class CommandIrisPregen extends MortarCommand {
```
Comment (cyberpwnn) [inline src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregen.java:28]: You need to register this in CommandIris.class

File: src/main/java/com/volmit/iris/core/command/world/CommandIrisPregen.java [hunk 5]
```diff
@@ -1,185 +0,0 @@
...
-import com.volmit.iris.util.math.Position2;
-import com.volmit.iris.util.plugin.MortarCommand;
-import com.volmit.iris.util.plugin.VolmitSender;
-import org.bukkit.Bukkit;
-import org.bukkit.World;
-import org.bukkit.entity.Player;
-
-import java.awt.*;
-
-public class CommandIrisPregen extends MortarCommand {
```
Comment (cyberpwnn) [inline src/main/java/com/volmit/iris/core/command/world/CommandIrisPregen.java:36]: You need to unregister this in CommandIris.class

File: src/main/java/com/volmit/iris/core/pregenerator/PregenTask.java [hunk 6]
```diff
@@ -34,12 +34,15 @@
...
+    private int width = 1;
+
+    @Builder.Default
+    private int height = 1;
 
     private static final KList<Position2> order = computeChunkOrder();
 
     public void iterateRegions(Spiraled s) {
-        new Spiraler(radius * 2, radius * 2, s)
+        new Spiraler(getWidth() * 2, getHeight() * 2, s)
```
Comment (cyberpwnn) [inline src/main/java/com/volmit/iris/core/pregenerator/PregenTask.java:45]: Just go back to a radius for now please

File: src/main/java/com/volmit/iris/core/pregenerator/PregenTask.java [hunk 7]
```diff
@@ -50,7 +53,7 @@ public static void iterateRegion(int xr, int zr, Spiraled s) {
     }
 
     public void iterateAllChunks(Spiraled s) {
-        new Spiraler(radius * 2, radius * 2, (x, z) -> iterateRegion(x, z, s))
+        new Spiraler(getWidth() * 2, getHeight() * 2, (x, z) -> iterateRegion(x, z, s))
```
Comment (cyberpwnn) [inline src/main/java/com/volmit/iris/core/pregenerator/PregenTask.java:56]: please

**Review by cyberpwnn — CHANGES_REQUESTED** [review]
Some issues

File: src/main/java/com/volmit/iris/Iris.java [hunk 8]
```diff
@@ -267,8 +268,8 @@ public void onDisable() {
             proj.close();
 
             for (World i : Bukkit.getWorlds()) {
```
Comment (cyberpwnn) [inline src/main/java/com/volmit/iris/Iris.java:269]: This is unrelated to Pregen, make seperate PRs on SEPERATE BRANCHES on your repo, otherwise, two PRs trying to merge the same branch at different times will still have the newest changes.

File: src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregenCreate.java (hunk 1 above)
Comment (svdgoor) [reply src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregenCreate.java]: You got it

File: src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregenCreate.java (hunk 2 above)
Comment (svdgoor) [reply src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregenCreate.java]: You can still do /iris pregen create 5000
Or /iris pregen create radius=5000
Same thing.
Tried the width&height with different values and it seems to work
I can remove them from the description and args usage but keep functionality (hiding them but still allowing usage), or I can just keep them, up 2 u.

File: src/main/java/com/volmit/iris/core/command/world/CommandIrisPregen.java (hunk 5 above)
Comment (svdgoor) [reply src/main/java/com/volmit/iris/core/command/world/CommandIrisPregen.java:36]: I'll remove the class

**Comment by svdgoor** [comment]
Apologies for the combined PR. I accidentally branched off of my branch for the toolbelt thing.

File: src/main/java/com/volmit/iris/Iris.java (hunk 8 above)
Comment (svdgoor) [reply src/main/java/com/volmit/iris/Iris.java:269]: yup

File: src/main/java/com/volmit/iris/core/command/world/CommandIrisPregen.java (hunk 5 above)
Comment (svdgoor) [reply src/main/java/com/volmit/iris/core/command/world/CommandIrisPregen.java:36]: Unsure why this comment is here, I moved the class before making the changes (refactor) so all imports should redirect properly. I don't see any reference to .iris.core.command.world.CommandIrisPregen

**Comment by svdgoor** [comment]
TLDR;
Renamed commands as requested
Prevented some overriding issues & improved debugging for pregen start
`radius=` is now the only "visible" size argument, but `width=` and `height=` are still internally possible.
